### PR TITLE
feat: Reduce CRD size

### PR DESCRIPTION
**Pull request by sbernauer**
# Description

Before 2.4M
After operator-rs bump (to pull in https://github.com/stackabletech/operator-rs/pull/821): 289K
After extraVolumes schema change: 183K :rocket: 

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes

```[tasklist]
# Author
- [x] Changes are OpenShift compatible
- [x] CRD changes approved
- [x] CRD documentation for all fields, following the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs-style-guide).
- [x] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
- [x] Changes need to be "offline" compatible
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] Code contains useful logging statements
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated. Follows the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs-style-guide).
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
- [ ] [Roadmap](https://github.com/orgs/stackabletech/projects/25/views/1) has been updated
```

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -31,5 +31,6 @@ strum = { version = "0.26", features = ["derive"] }
 tokio = { version = "1.37", features = ["full"] }
 tracing = "0.1"
 
-# [patch."https://github.com/stackabletech/operator-rs.git"]
-# stackable-operator = { git = "https://github.com/stackabletech//operator-rs.git", branch = "main" }
+[patch."https://github.com/stackabletech/operator-rs.git"]
+# stackable-operator = { path = "/home/sbernauer/stackable/operator-rs/crates/stackable-operator" }
+stackable-operator = { git = "https://github.com/stackabletech//operator-rs.git", branch = "feat/reduce-crd-size" }
```
Comment (sbernauer) [inline Cargo.toml:36]: marker to change back once releaed

**Comment by NickLarsenNZ** [comment]
I'm curious what the CRD docs would look like. Can we link to somewhere for the rest of the schema? Do we generate two schemas, a full (for docs) and limited (to apply to k8s)?

Although the end-user loses validation directly from k8s-api, the operator will not accept an invalid CR. If we want to shift validation left, we can then look at validating webhooks.

And as you also mentioned, hopefully some day we can bring it back in if Kubernetes handles references to sub-schemas.

**Comment by NickLarsenNZ** [comment]
Just want to make sure @nightkr sees this, before we continue.
Related PR: https://github.com/stackabletech/operator-rs/pull/821

**Comment by sbernauer** [comment]
Please vote if you agree with this change :+1: / :-1: , I personally don't see any way around this when we want to have CRD versioning

**Comment by nightkr** [comment]
_Ideally_ I would like to keep it for the latest version even if we prune it for the older..

**Comment by sbernauer** [comment]
> Ideally I would like to keep it for the latest version even if we prune it for the older..

That's an interesting idea, thanks! Lets say we start with v1 (full). Now we bump and have v2 (full) and v1 (without PodTemplate schema). By doing so we would alter the schema of v1 after the fact - does that cause any harm? It sounds dangerous to me, but it might be totally safe. But It might not fit the mental model of the stability of CRD versions :thinking:

**Review by labrenbe — APPROVED** [review]
lgtm

**Comment by sbernauer** [comment]
Started https://testing.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/druid-operator-it-custom/10/

**Comment by sbernauer** [comment]
https://testing.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/druid-operator-it-custom/11/

**Review by maltesander — APPROVED** [review]
LGTM!

**Pull request merged**